docs/niscope/functions.rst

```diff
@@ -1212,9 +1212,18 @@ niscope.Session methods
 
         .. code:: python
 
-            session.channels['0,1'].fetch(num_samples, timeout='datetime.timedelta(seconds=5.0)')
+            session.channels['0,1'].fetch(timeout='datetime.timedelta(seconds=5.0)', num_samples=None, relative_to=niscope.FetchRelativeTo.PRETRIGGER, offset=0, record_number=0, num_records=-1)
```

Comment:
Since num_samples=None is the default and means "you choose for me", why not make num_records the same?

Reply:
The difference is that `None` means go to the driver and get a number that I use to allocate memory, and the -1 is passes into the driver as is.

It would make the logic look like this:
```python
self._fetch_num_records = -1 if num_records is None else num_records
if num_samples is None:
    num_samples = self.horz_record_length
```

Is that okay?

Reply:
I think that's ok. The important thing is the interface.

Reply:
On one hand, I think public API is what matters. On the other hand, we have fully generated functions in which -1 means "automatic". I don't have a strong opinion either way, as long as docstring is helpful.